br/br-snapshot-architecture.md

```diff
@@ -0,0 +1,160 @@
+
+## Process of restore
+
+The process of a cluster snapshot restore is as follows:
+
+![snapshot restore process design](/media/br/br-snapshot-restore-ts.png)
+
+The complete restore process is as follows:
+
+1. `br` receives the `br restore` command.
+
+    * Gets the data storage path and the database or table to be restored.
+    * Checks whether the table to be restored exists and whether it meets the requirements for restore.
+
+2. `br` schedules the restore data.
+
+    * **Pause Region schedule**: `br` requests PD to pause the automatic Region scheduling during restore.
+    * **Restore schema**: `br` gets the schema of the backup data and the database and table to be restored. Note that the ID of a newly created table might be different from that of the backup data.
+    * **Split & scatter Region**: `br` requests PD to allocate Regions (split Region) based on backup data, and schedules Regions to be evenly distributed to storage nodes (scatter Region). Each Region has a specified data range `[start key, end key)`.
+    * **Request TiKV to restore data**: `br` creates a restore request and sends it to the corresponding TiKV nodes according to the result of Region split. The restore request includes the data to be restored and rewrite rules.
+
+3. TiKV accepts the restore request and initiates a restore worker.
+
+    * The restore worker calculates the backup data that needs to be read to restore.
+
+4. TiKV restores the data.
+
+    * **Download SST**: the restore worker downloads corresponding SST files from the storage path to a local directory.
+    * **Rewrite KVs**: the restore worker rewrites the KV data according to the new table ID, that is, replace the original table ID in the [Key-Value](/tidb-computing.md#mapping-table-data-to-key-value) with the new table ID. The restore worker also rewrites the index ID in the same way.
+    * **Ingest SST**: the restore worker ingests the processed SST files into RocksDB.
+    * **Report restore result**: the restore worker reports the restore result to `br`.
+
+5. `br` receives the restore result from each TiKV node.
+
+    * If some data fails to be restored due to `RegionNotFound` or `EpochNotMatch`, for example, a TiKV node is down, `br` will retry the restore.
+    * If there is any data fails to be restored and cannot be retried, the restore task fails.
+    * After all data is restored, the restore task succeeds.
+
+For more details about the backup and restore process, see [Design of backup and restore](https://github.com/pingcap/tidb/blob/master/br/docs/cn/2019-08-05-new-design-of-backup-restore.md).
+
+## Backup files
+
+### Types of backup files
+
+Snapshot backup generates the following types of files:
+
+- `SST` file: stores the data that the TiKV node backs up. The size of an `SST` file equals to that of a Region.
+- `backupmeta` file: stores the metadata of a backup task, including the number, the key range, the size, and the Hash (sha256) value of the backup files.
```

Comment:
`the key range` is not an attribution of `backup files`, it should be that backup files contains the key range be backuped.